packages/ui/src/components/Node/ItemCollection.test.ts

```diff
@@ -0,0 +1,110 @@
+import { describe, it, expect } from 'vitest';
+import { ItemCollection } from './ItemCollection';
+import { nested } from './itemExamples/nested';
+
+describe('toTable', () => {
+  it ('should correctly extract headers and rows from nested array', () => {
+    const mockData = [
+      {
+        'foo': 'bar',
+        'baz': [
+          {
+            'foo': 'bar',
+            'baz': 'qux'
+          }
+        ]
+      }
+    ];
+
+    const { headers, rows } = new ItemCollection(mockData).toTable();
+    expect(headers).toMatchInlineSnapshot(`
+      [
+        "foo",
+        "baz",
+      ]
+    `);
+    expect(rows).toMatchInlineSnapshot(`
```

Comment:
In general, and in this case, are there reasons to use snapshot over [toMatchObject](https://vitest.dev/api/expect#tomatchobject)? We loose IDE help when its just a string so I think prefer toMatchObject

Reply:
This case passes: `expect({foo:'hi', bar: 'hello'}).toMatchObject({bar: 'hello'});` 

using `toMatchObject` when we're only interested in checking if certain properties of the object match our expectations. 
 In this scenario, I'm looking to test if the entire result meets the expectations.

Reply:
Got it! Can we use toEqual for those cases? 

```ts
  it('can use toEqual even on nested objects', () => {
    const a = { foo: { bar: { baz: 'qux' } } };
    const b = { foo: { bar: { baz: 'qux' } } };
    expect(a).toEqual(b);
  });
```
This passes

Reply:
Seeing the update👍 No downside with this approach right? Lets merge when you are ready